### ISSUE TITLE
Question/Bug? - Are arangosearch views within transactions supported in 3.7?

### ISSUE THREAD

**Issue by dvstans** [arangodb/arangodb#12949]
Before I log a bug, I wanted to check and see if this may be a known issue or limitation with 3.7.x (I couldn't find any info when searching). I recently migrated from 3.6.1 to 3.7.3 and all of my Foxx services that use aragosearch view-based queries from inside a transaction now fail with the following error:

"Unexpected DB exception: ArangoError 4: AQL: failed to get snapshot while creating arangosearch view ExecutionBlock (exception location: /work/ArangoDB/arangod/Aql/IResearchViewNode.cpp:1473). Please report this error to arangodb.com (while optimizing plan) (exception location: /work/ArangoDB/arangod/V8Server/v8-vocbase.cpp:162). Please report this error to arangodb.com"

If I run the same queries outside of a transaction, they work as expected. I'll provide full information if this is a new issue.

**Comment by maxkernbach** [comment]
Hi @dvstans,

Thanks a lot for reporting, this seems to be a bug indeed.

We will look into this issue and keep you updated. No further information from your side is needed for now, we were able to reproduce the same error.

**Comment by Yogu** [comment]
Did you find a reason for this error?

We also get this error with our library in 3.7: https://github.com/AEB-labs/cruddl/pull/182/checks?check_run_id=1856288368

**Comment by CBlinguland** [comment]
Hi,
today I got the same error - using a view on ArangoDB 3.7.9 (Linux).

My workaround/solution:
I added the name of view AND the collection name used in the definition of view as read-collection.

My view is named **find_user** and it uses collection **user**. So i defined these name as read-collection for the transaction.

db._executeTransaction({
	"collections": {
		"read": ["find_user","user"],
		"write": [],
		"allowImplicit": true
	},
	"params": {},
	...
});

Now running find. Don't remember if it was documented...

Good luck
